pkg/storage/bucket/client.go

```diff
@@ -125,6 +127,10 @@ func NewClient(ctx context.Context, cfg Config, name string, logger log.Logger,
 		return nil, err
 	}
 
+	if cfg.StoragePrefix != "" {
```

Comment:
I suggest restricting the character set allowed in the storage prefix to the ones listed as safe characters in the [S3 docs](https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-keys.html#object-key-guidelines). It can save a lot of headaches and subtle bugs.

Reply:
nice idea, I pushed a change, while still working on the rest of the tests